doc/sag/sag-HLD.md

```diff
@@ -0,0 +1,185 @@
+# SAG high level design for SONiC
+
+# Table of Contents
+- [SAG high level design for SONiC](#sag-high-level-design-for-sonic)
+- [Table of Contents](#table-of-contents)
+- [Revision History](#revision-history)
+- [Abbreviations](#abbreviations)
+- [Requirements Overview](#requirements-overview)
+- [Modules Design](#modules-design)
+  - [Config DB](#config-db)
+    - [SAG Global Table](#sag-global-table)
+    - [SAG Interface Table](#sag-interface-table)
+    - [Config DB Schemas](#config-db-schemas)
+  - [APP DB](#config-db)
+    - [Interface Table](#interface-table)
+    - [APP DB Schemas](#app-db-schemas)
+  - [Orchestration Agent](#orchestration-agent)
+    - [SagOrch](#sagorch)
+    - [SagMgrD](#sagmgrd)
+    - [IntfsOrch](#intfsorch)
+  - [SAI](#sai)
+- [Flows](#flows)  
+- [CLI](#cli)
+- [Test](#test)
+  - [Functional Test Cases](#functional-test-cases)
+
+# Revision History
+|  Rev  | Date       |   Author    | Change Description                |
+| :---: | :--------- | :---------: | --------------------------------- |
+|  0.1  | 08/13/2021 |  Jimi Chen  | Initial version                   |
+
+# Abbreviations
+| Abbreviations | Description            |
+|---------------|------------------------|
+| SAG           | Static Anycast Gateway |
+
+# Requirements Overview
+Anycast gateway feature for EVPN/VXLAN is a default gateway addressing mechanism that enables the use of the same gateway IP address across all the leaf switches that are part of a VXLAN network.  
+This ensures that every leaf switch can function as the default gateway for the workloads directly connected to it.  
+The feature failitates flexible workload placement, host mobility and optimal traffic forwarding across the VXLAN fabric.  
+
+This feature should be used together with VxLAN EVPN.  
+It should always to configure VxLAN to make SAG work.  
+SAG announces its IP and MAC only towards the host facing Ethernet ports but not towards the fabric (Overlay).  
+As a result, the IP/MAC for the SAG is only seen and reachable from the host facing side and never from the fabric (Overlay) facing side.
+
```

Comment:
Please explain how this anycast gateway IP is not advertised to the fabric.

Reply:
Update the description in the HLD, it can be deployed individually without EVPN.  
In common use case in VxLAN/EVPN, SAG is configured towards to the host (which is overlay network), and it should not advertised to the fabric side (which is underlay network).